crates/hdi/src/op/test.rs

```diff
@@ -0,0 +1,205 @@
+use super::*;
+use crate as hdi;
+use crate::test_utils::set_zome_types;
+use crate::test_utils::short_hand::*;
+use test_case::test_case;
+
+#[hdk_entry_helper]
+#[derive(Clone, PartialEq, Eq)]
+pub struct A;
+#[hdk_entry_helper]
+#[derive(Clone, PartialEq, Eq)]
+pub struct B;
+#[hdk_entry_helper]
+#[derive(Clone, PartialEq, Eq)]
+pub struct C;
+
+#[hdk_entry_helper]
+#[derive(Clone, PartialEq, Eq, Default)]
+pub struct D {
+    a: (),
+    b: (),
+}
+
+#[hdk_entry_defs(skip_hdk_extern = true)]
+#[unit_enum(UnitEntryTypes)]
+#[derive(Clone, PartialEq, Eq)]
+pub enum EntryTypes {
+    A(A),
+    #[entry_def(visibility = "private")]
+    B(B),
+    C(C),
+}
+#[hdk_link_types(skip_no_mangle = true)]
+pub enum LinkTypes {
```

Comment:
same as above we could use `num_enum` here for readability

Reply:
asking because i saw this:

https://github.com/holochain/integrity-template/blob/4bbccf86aa0ebb08b4a5a24e34bdefaa80ca121e/dna/zomes/mydna1_core/src/lib.rs#L57-L65

Reply:
Yeh that is wrong. I'm not sure where that came from.

Reply:
@zippy please shed some light on where you got that pattern from